swingset/src/main/java/com/nqadmin/swingset/SSComboBox.java

```diff
@@ -914,6 +987,57 @@ public void setSelectedValue(final Integer _value) {
 
 	// }
 
+	private void adjustForNullItem() {
+		try (Model.Remodel remodel = optionSwingModel.getRemodel()) {
+			boolean wantNull = getAllowNull();
+			boolean hasNull = nullItem != null;
+
+			if (wantNull == hasNull) {
+				// things are set up as wanted
+				return;
+			}
+			
+			// Need to add/remove the nullItem
+			if (wantNull) {
+				nullItem = remodel.createComboItem(null, "", null);
+				remodel.add(0, nullItem);
+			} else {
+				remodel.remove(nullItem);
+				nullItem = null;
+			}
+		}
+	}
+
+	/**
+	 * After this, make some adjustments.
+	 * {@inheritDoc }
+	 */
+	@Override
+	public void setAllowNull(boolean _allowNull) {
+		ssCommon.setAllowNull(_allowNull);
+		adjustForNullItem();
+	}
+
+	/**
+	 * After this, make some adjustments.
+	 * {@inheritDoc }
+	 */
+	//
+	// TODO: this never gets called
+	//
+	@Override
+	public void bind() {
+		SSComponentInterface.super.bind();
+		adjustForNullItem();
+	}
+
+	@Override
+	public void bind(RowSet _rowSet, String _boundColumnName) {
+		SSComponentInterface.super.bind(_rowSet, _boundColumnName);
+		adjustForNullItem();
+	}
```

Comment:
Need to catch metadata changes. Does this seem reasonable, in my tests bind() didn't get called, but bind(rs, name) did. Is there something else I should catch?

Reply:
Empty bind() didn't make sense at first, but I guess the idea was to set the pieces (column, rowset) and then bind. I made SSCommon.bind() private and commented it out of SSComponentInterface and no compile errors.

I found it in 3 spots in our app code, but just in comments referring conceptually to bind() so we can probably persist the changes to SSCommon and SSComponentInterface().

SSDataGrid and a few deprecated formatting subpackage components roll their own bind.

Reply:
> guess the idea was to set the pieces (column, rowset) and then bind

What does this capability contribute? Feels like allowing setting of mappings/options separately; it just opens the door to bugs and maintenance problems for no good reason.

Get rid of it?

Reply:
Yes please.